src/main/java/com/conveyal/datatools/manager/controllers/api/UserController.java

```diff
@@ -167,24 +182,56 @@ private static String createPublicUser(Request req, Response res) {
     }
 
     /**
-     * HTTP endpoint to create new Auth0 user for the application.
+     * HTTP endpoint to create new Auth0 user for the application. If the Auth0 user already exists, update the user
+     * permissions instead.
      */
     private static String createUser(Request req, Response res) {
+        JsonNode jsonNode = parseJsonFromBody(req);
+        String email = jsonNode.get("email").asText();
+        Auth0UserProfile user = checkForExistingAuth0User(email);
+
+        // User already exists, update permissions
+        if (user != null) {
+            String json = String.format("{ \"app_metadata\": { \"datatools\" : %s  }}", jsonNode.get("permissions"));
+            return updateUserPermissions(req, user, json);
+        }
+
+        // Create new user
         HttpPost createUserRequest = new HttpPost(baseUsersUrl);
         setHeaders(req, createUserRequest);
-
-        JsonNode jsonNode = parseJsonFromBody(req);
         String json = String.format("{" +
                 "\"connection\": \"Username-Password-Authentication\"," +
                 "\"email\": %s," +
                 "\"password\": %s," +
                 "\"app_metadata\": {\"datatools\": [%s] } }"
                 , jsonNode.get("email"), jsonNode.get("password"), jsonNode.get("permissions"));
         setRequestEntityUsingJson(createUserRequest, json, req);
-
```

Comment:
This section of the code has always been a bit messy with all of the JSON + escaped chars. It might be worth constructing an ObjectNode to create the JSON.

Reply:
Decided to follow the approach of the other JSON method by using the request and `logMessageAndHalt`

Reply:
@landonreed there many be a reason for this. The permissions return from the UI are already raw JSON, placing this into an ObjectNode escapes the quotes and produces the following mess. I don't think there is a way around this. Where possible I have used ObjectNodes but the impact is now minimal.
```

{
  "datatools": [
    "{\"permissions\":[],\"projects\":[{\"project_id\":\"77ef6322-79b1-4c44-afc3-5d767b2b9f02\",\"permissions\":[{\"type\":\"administer-project\"}]}],\"organizations\":[],\"client_id\":\"MPXcM4ayjpQhyVKXbgQ5oXAeixpPqHd2\"}"
  ]
}
```